optuna/samplers/_search_space/intersection.py

```diff
@@ -59,20 +59,20 @@ def calculate(self, study: Study, ordered_dict: bool = False) -> Dict[str, BaseD
             if self._study_id != study._study_id:
                 raise ValueError("`IntersectionSearchSpace` cannot handle multiple studies.")
 
-        states_of_interest = [optuna.trial.TrialState.COMPLETE]
+        states_of_interest = [optuna.trial.TrialState.COMPLETE, optuna.trial.TrialState.RUNNING]
 
         if self._include_pruned:
             states_of_interest.append(optuna.trial.TrialState.PRUNED)
 
-        next_cursor = self._cursor
-        for trial in reversed(study.get_trials(deepcopy=False)):
+        trials = study.get_trials(deepcopy=False, states=states_of_interest)
+
+        next_cursor = trials[-1].number if len(trials) > 0 else -1
```

Comment:
I think that the cursor should point to the youngest unfinished trial.
```suggestion
        next_cursor = trials[-1].number + 1 if len(trials) > 0 else -1
```

Reply:
Thanks, I updated the code.